### PR TITLE
Add unit tests for SnorkelClassifier.predict(), score()

### PR DESCRIPTION
**Pull request by bhancock8**
- Adds basic tests for the predict() and score() methods
- Also fixes a bug discovered by the unit tests (yay unit tests!) when calculating abstain mask
- Moves filtering of examples with unknown gold labels to the metrics instead of the predict() method so that predict returns a label for every example in the dataset.

**Test plan:**
Adds unit test confirming that predict() and score() yield outputs of the right types and shapes.

### CODE REVIEW

File: test/classification/test_snorkel_classifier.py [hunk 1]
```diff
@@ -2,38 +2,83 @@
...
+        name=task_name,
+        module_pool=module_pool,
+        task_flow=task_flow,
+        scorer=Scorer(metrics=["accuracy"]),
+    )
+
+    return task
+
+
+task1 = create_task("task1", module_suffixes=["A", "A"])
```
Comment (henryre) [inline test/classification/test_snorkel_classifier.py]: Nit: caps in global scope

Comment (henryre) [reply test/classification/test_snorkel_classifier.py]: or make these attributes in setUp

File: test/classification/test_snorkel_classifier.py [hunk 2]
```diff
@@ -45,12 +90,37 @@ def test_twotask_partial_overlap_model(self):
...
+        self.assertEqual(sorted(list(results.keys())), ["golds", "preds", "probs"])
+        self.assertEqual(results["preds"]["task1"].shape, (NUM_EXAMPLES,))
+
+    def test_empty_batch(self):
+        # Make the first BATCH_SIZE labels 0 so that one batch will have no labels
+        dataloader = create_dataloader("task1", shuffle=False)
+        for i in range(BATCH_SIZE):
+            dataloader.dataset.Y_dict["task1"][i] = 0
+        model = SnorkelClassifier([task1])
+        model.predict(dataloader)
```
Comment (henryre) [inline test/classification/test_snorkel_classifier.py]: What do we expect this to be?

File: test/classification/test_snorkel_classifier.py [hunk 3]
```diff
@@ -45,12 +90,37 @@ def test_twotask_partial_overlap_model(self):
...
+
+    def test_score(self):
+        model = SnorkelClassifier([task1])
+        metrics = model.score([dataloader])
+        self.assertIsInstance(metrics["task1/dataset/train/accuracy"], float)
```
Comment (henryre) [inline test/classification/test_snorkel_classifier.py]: What should the accuracy be?

File: test/classification/test_snorkel_classifier.py [hunk 4]
```diff
@@ -2,38 +2,83 @@
...
+    return dataloader
+
+
+def create_task(task_name, module_suffixes=("", "")):
+    module1_name = f"linear1{module_suffixes[0]}"
+    module2_name = f"linear2{module_suffixes[1]}"
+
+    module_pool = nn.ModuleDict(
+        {
+            module1_name: nn.Sequential(nn.Linear(2, 4), nn.ReLU()),
```
Comment (henryre) [inline test/classification/test_snorkel_classifier.py]: Is it worth it to manually set the weights so we know expected outputs of a forward pass? maybe just to the identity?

File: test/classification/test_snorkel_classifier.py (hunk 3 above)
Comment (bhancock8) [reply test/classification/test_snorkel_classifier.py]: Unknown--it's a randomly initialized network.

File: test/classification/test_snorkel_classifier.py (hunk 4 above)
Comment (bhancock8) [reply test/classification/test_snorkel_classifier.py]: Hmm, yeah. I'll give it a try.

**Comment by bhancock8** [comment]
Ready for another look @henryre

File: snorkel/analysis/metrics.py [hunk 5]
```diff
@@ -18,7 +18,7 @@ def metric_score(
     preds: np.ndarray,
     probs: np.ndarray,
     metric: str,
-    filter_dict: Optional[Dict[str, List[int]]] = None,
+    filter_dict: Dict[str, List[int]] = {"golds": [0]},
```
Comment (henryre) [inline snorkel/analysis/metrics.py]: No mutables for defaults plz (https://docs.python-guide.org/writing/gotchas/#mutable-default-arguments)

Note to self: this isn't caught by flake8, but we can with an add-on https://github.com/pycqa/flake8-bugbear
